bin/opd8n/src/cmd/from_l2.rs

```diff
@@ -18,6 +48,231 @@ pub struct FromL2 {
+            "End block before start block"
+        );
+        trace!(target: TARGET, "Producing derivation fixture for L2 block range [{}, {}]", self.start_block, self.end_block);
+
+        // Build the pipeline
+        let cfg = Arc::new(self.rollup_config().await?);
+        let mut l1_provider = self.l1_provider()?;
+        let mut l2_provider = self.l2_provider(cfg.clone())?;
+        let attributes = self.attributes(cfg.clone(), &l2_provider, &l1_provider);
+        let mut blob_provider = self.blob_provider();
+        let dap = self.dap(l1_provider.clone(), blob_provider.clone(), &cfg);
+        let mut l2_cursor = self.cursor().await?;
+        let l1_tip = l1_provider
+            .block_info_by_number(l2_cursor.l1_origin.number)
+            .await
+            .expect("Failed to fetch genesis L1 block info for pipeline tip");
+        let mut pipeline = new_online_pipeline(
+            cfg.clone(),
+            l1_provider.clone(),
+            dap,
+            l2_provider.clone(),
+            attributes,
+            l1_tip,
+        );
+
+        // Collect the output payload attributes.
+        let mut payloads = Vec::with_capacity((self.end_block - self.start_block) as usize);
+
+        // Track the earliest l1 block and last l1 block.
+        let first_l1_block = l2_cursor.l1_origin.number;
+        let mut last_l1_block = l2_cursor.block_info.number;
+
+        // Run the pipeline
+        loop {
+            // If the cursor is beyond the end block, break the loop.
+            if l2_cursor.block_info.number >= self.end_block {
+                trace!(target: TARGET, "Cursor is beyond the end block, breaking loop");
+                break;
+            }
+
+            // Step on the pipeline.
+            match pipeline.step(l2_cursor).await {
+                StepResult::PreparedAttributes => trace!(target: "loop", "Prepared attributes"),
+                StepResult::AdvancedOrigin => trace!(target: "loop", "Advanced origin"),
+                StepResult::OriginAdvanceErr(e) => {
+                    warn!(target: TARGET, "Could not advance origin: {:?}", e)
+                }
+                StepResult::StepFailed(e) => match e {
+                    StageError::NotEnoughData => {
+                        debug!(target: TARGET, "Not enough data to step derivation pipeline");
+                    }
+                    _ => {
+                        error!(target: TARGET, "Error stepping derivation pipeline: {:?}", e);
+                    }
+                },
+            }
+
+            // Get the attributes if there are some available.
+            let Some(attributes) = pipeline.next() else {
+                continue;
+            };
+
+            // Print the L1 range for this L2 Block.
+            let derived = attributes.parent.block_info.number as i64 + 1;
+            let l2_block_info = l2_provider
+                .l2_block_info_by_number(derived as u64)
```

Comment:
Is there a reason you are casting `derived` as `i64` and then as `u64` directly after? Is there a reason `derived` needs to be `i64`?